flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -431,6 +433,29 @@ public Task(
 
         // finally, create the executing thread, but do not start it
         executingThread = new Thread(TASK_THREADS_GROUP, this, taskNameWithSubtask);
+
+        // add CallerContext.
+        String callerContext =
+                "Flink_Task_"
+                        + "JobID_"
+                        + jobId
+                        + "_TaskName_"
+                        + taskInfo.getTaskName()
+                        + "_"
+                        + taskInfo.getAttemptNumber();
+
+        // First check if we have Hadoop in the ClassPath. If not, we simply don't do anything.
+        try {
```

Comment:
We shouldn't be leaking any hadoop related code into the Task implementation. Ideally this change-set would be only local to the `flink-hadoop-fs` module.

CallerContext is implemented using (inheritable) thread local, these are pretty cheap to access. Can we simply re-set the context any time we're accessing the FileSystem?